grid.v3/client.go

```diff
@@ -29,11 +29,7 @@ type Client struct {
 }
 
 // NewClient with namespace and using the given etcd client.
-func NewClient(etcd *etcdv3.Client, namespace string) (*Client, error) {
-	r, err := registry.New(etcd)
-	if err != nil {
-		return nil, err
-	}
+func NewClient(etcd *etcdv3.Client, r *registry.Registry, namespace string) (*Client, error) {
```

Comment:
I don't know how I feel about needing the user to pass in the registry...

What's so compelling about passing it in?